src/kudu/consensus/consensus_peers.cc

```diff
@@ -208,6 +212,9 @@ Status Peer::SignalRequest(bool even_if_queue_empty) {
 
 void Peer::SendNextRequest(bool even_if_queue_empty) {
   std::unique_lock<simple_spinlock> l(peer_lock_);
+
+  last_request_time_ = MonoTime::Now();
+
```

Comment:
if (request_pending_ || (!even_if_queue_empty && using_proxy && (MonoTime::Now() - last_request_time_ < FLAGS_proxy_batch_duration_ms) {
   return;
}

Reply:
Moved all logic to SendNextRequest(). I thought about caching the proxy state in the peer obj but I don't think the routing table lookup should be very costly, so went with just calling it for now and added a TODO. Maintaining state in peer obj will be a slightly bigger change.